app/views/pages/_coc_text.html.erb

```diff
@@ -1,7 +1,11 @@
 <p>All participants of <%= community_name %> are expected to abide by our Code of Conduct, both online and during in-person events that are hosted and/or associated with <%= community_name %>.</p>
 <h2>Our Pledge</h2>
 <p>In the interest of fostering an open and welcoming environment, we as moderators of
+<% if FeatureFlag.enabled?(:creator_onboarding) && User.with_role(:creator).any? && request.env["PATH_INFO"] == new_admin_creator_setting_path %>
+  <%= community_name %> pledge to make participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, gender identity and expression, level of experience, nationality, personal appearance, race, religion, or sexual identity and orientation.
+<% else %>
   <a href="<%= app_url %>"><%= community_name %></a> pledge to make participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, gender identity and expression, level of experience, nationality, personal appearance, race, religion, or sexual identity and orientation.
+<% end %>
```

Comment:
This conditional and the two below ensure that links do not link through in the COC and TOS modals (the links will not work, as they redirect off of the Creator Settings form and we have checks in place to ensure that this is not allowed _until_ a Creator completes onboarding. See `onboardingRedirectCheck.jsx` for the code that handles this redirect check.), but do link through on any other page, such as on the COC and TOS links in the lefthand sidebar on the homepage.

Reply:
Given the repetition of three bits of logic it would be good to either extract a helper method or set a variable at the top of the template:

```erb
<% show_creator_onboarding = FeatureFlag.enabled?(:creator_onboarding) && User.with_role(:creator).any? && request.env["PATH_INFO"] == new_admin_creator_setting_path %>
```

That helps keep the repetition of logic synchronized.  And you could add a comment to help clarify the context (what you have in the above comment, for example)

Reply:
Agreed and done, @jeremyf!